sdk/keyvault/TEST_RESOURCES_README.md

```diff
@@ -0,0 +1,130 @@
+
+## The form to deploy the KeyVault resources
+
+Once you click the "Deploy to Azure" button, you will be prompted to fill a form.
+In this guide we will go through the necessary steps to fill this form for the KeyVault clients for JavaScript.
+
+### Logging in
+
+Clicking the button will make Azure attempt to authenticate your Azure account. Make sure to log in using the Azure account that you want to be billed through.
+If you want to **create a new account**, you may follow this guide: [Create an Azure account][CreateAzureAccount].
+
+### The "Subscription" field
+
+The first field you will encounter is the "Subscription" field.
+It will have a drop down menu where you will be able to search for the specific Azure subscription
+you may want to use as the host of the resources you will create.
+
+**This field is required.**
+
+### The "Resource Group" field.
+
+Having picked a subscription, the resource groups available for that subscription
+will be pre-filled in a drop-down menu. You can either pick one resource group from this menu,
+or click on the "Create new" link right under the drop-down menu to create a new resource group.
+You can read more about Resource groups [here][ResourceGroups].
+
+**This field is required.**
+
+### The "Location" field.
+
+Azure has more global regions than any other cloud provider.
+Once you've picked a subscription and a resource group,
+you'll be able to pick a location. 
+You can read more about Azure locations [here][AzureLocations].
+
+**This field is required.**
+
+### The "Base Name" field.
+
+Under "Settings", the "Base Name" will serve us as a prefix
+to determine the name of the resources that we will create,
+so that you can easily find them later.
+
+This field gets automatically generated for us through our
+CI pipelines. It is **required** to replace this default value
+with any combination of alphanumeric characters that you might
+prefer. Keep in mind that if you have previously generated
+resources with the same alphanumeric combination that you're
+providing, Azure will likely fail, requiring you to insert
+a different value.
+
+### The "Tenant Id" field.
+
+To fill the **Tenant Id** field, you may copy it from the one available on the
+[overview page of the Azure Active Directory section of the Azure portal][AADOverview],
+or obtain it from the overview page of a recently created Azure Application.
+Here's [how to use the portal to create an Azure AD application and service principal that can access resources][HowToAADApp].
+
+**This field is required.**
+
+### The "Test Application Oid" field.
+
+To fill the **Test Application Oid** field, you may get it from the overview page of a recently created Azure application.
+
+If you don't have an Azure AD application, follow: [How to use the portal to create an Azure AD application and service principal that can access resources][HowToAADApp].
```

Comment:
I pressed things on the keyboard 🤣 Nothing concrete! I'll pick a better name.

Reply:
Oh, so this is not the ADD Application ID, it's the Object ID of the AAD Application. I know, it's confusing for everyone.

Reply:
Some more things to think about (from our brainstorming chat earlier):
1. Use the tooltip (which appears to map to the { metadata: description } field from the ARM template.
2. We could also try embedding a link - this would give the user a way to easily find a more extensive set of instructions directly from the tooltip.

Reply:
There are 2 ways to access this GUID. 

**From Azure Active Directory -> App Registrations** 

This is where most people would probably land from the linked instructions.

1. Select the app whose Object ID you are looking for. Once you're on the "Overview" page, click the application name in the **Managed application in local directory** 

![image](https://user-images.githubusercontent.com/2158838/75466308-dd11d800-593e-11ea-88f0-885c7c0e8d85.png)


2. The Enterprise Application page for the app will load 
3. Select the Object ID from that page

![image](https://user-images.githubusercontent.com/2158838/75466112-86a49980-593e-11ea-9a35-e601a8c9acac.png)


**From Azure Active Directory -> Enterprise applications** 

1. Select the app whose Object ID you are looking for
2. From the "Overview" page select the Object ID (same as step 3 above)

![image](https://user-images.githubusercontent.com/2158838/75466112-86a49980-593e-11ea-9a35-e601a8c9acac.png)